### PR TITLE
boards/cc26xx_cc13xx-based: model features in Kconfig

### PR DESCRIPTION
**Pull request by jeandudey**
### Contribution description

This adds Kconfig symbol definitions for the `cc26xx_c13xx` CPUs and for boards based on that CPU.

### Testing procedure

- Green CI.
- `tests/kconfig_features` should pass.

### Issues/PRs references

#14148

### CODE REVIEW

File: tests/kconfig_features/Makefile [hunk 1]
```diff
@@ -1,6 +1,11 @@
 include ../Makefile.tests_common
 
-BOARD_WHITELIST += samr21-xpro
+BOARD_WHITELIST += samr21-xpro \
+                   cc1312-launchpad \
+                   cc1352-launchpad \
+                   cc1352p-launchpad \
+                   cc2650-launchpad \
+                   cc2650stk
```
Comment (leandrolanzieri) [inline tests/kconfig_features/Makefile:8]: Let's keep alphabetical order

File: cpu/cc26x0/Kconfig [hunk 2]
```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2020 Locha Inc
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config CPU_FAMILY_CC26X0
```
Comment (leandrolanzieri) [inline cpu/cc26x0/Kconfig:8]: I'm a bit confused here about the family and the series having the same name. Would it be OK to define a `CC26xx` series? I know the doc in `cpu/Kconfig` says that the family selects the core, but I think we need to change that, as sometimes it is selected by the series as well.

Comment (jeandudey) [reply cpu/cc26x0/Kconfig:8]: The CPU family in this case would be `CC26X0_CC13X0` which both use a Cortex-M3 (RIOT doesn't define a board for `cc13x0` however, they're compatible).

That would leave us with `CC26x0` series and when wanted a possible `CC13x0` series. Are you ok with it?

File: tests/kconfig_features/Makefile (hunk 1 above)
Comment (jeandudey) [reply tests/kconfig_features/Makefile]: Solved on https://github.com/RIOT-OS/RIOT/pull/14210/commits/50e460c984317398745c5833d6afaa6cf9499224

File: cpu/cc26x0/Kconfig (hunk 2 above)
Comment (leandrolanzieri) [reply cpu/cc26x0/Kconfig:8]: Yes, this would be similar to the symbols in `cc26x2_cc13x2` right?

Comment (jeandudey) [reply cpu/cc26x0/Kconfig:8]: Yes, same way

Comment (jeandudey) [reply cpu/cc26x0/Kconfig]: I've updated the family to `CC26X0_CC13X0`

File: cpu/cc26xx_cc13xx/Kconfig [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# Copyright (c) 2020 Locha Inc
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config CPU_CC26XX_CC13XX
```
Comment (leandrolanzieri) [inline cpu/cc26xx_cc13xx/Kconfig:8]: I understand that the `cc26xx_cc13xx` is just common code, not used directly by any board. I think in that case we can rename this symbol to `CPU_COMMON_CC26XX_CC13XX` to indicate that.

Comment (jeandudey) [reply cpu/cc26xx_cc13xx/Kconfig]: Makes sense, I've rename it to `CPU_COMMON_CC26XX_CC13XX`

**Comment by jeandudey** [comment]
All commits squashed and changes addressed. Thanks for the review!

File: cpu/cc26x0/Kconfig [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+#
+
+config CPU_FAMILY_CC26X0_CC13X0
+    bool
+    select CPU_CORE_CORTEX_M3
+    select CPU_CONFIG_CC26XX_CC13XX
```
Comment (leandrolanzieri) [inline cpu/cc26x0/Kconfig:11]:
```suggestion
    select CPU_COMMON_CC26XX_CC13XX
```

Comment (jeandudey) [reply cpu/cc26x0/Kconfig]: Oops, sorry, fixed!

**Pull request merged**